### DISCUSSION TITLE
Extra Element ID but for nodes?

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#25219] [Q&A Meshing] [answered]
Is this possible? Would be really convenient for defining field data for nodal auxkernels.

**Comment by GiudGiud** [comment]
@roystgnr

**Comment by roystgnr** [comment, marked as answer]
It's there in the libMesh internals; just `MeshBase::add_node_integer` instead of `add_elem_integer` (or `add_node_datum` to add something other than `dof_id_type`, or `add_node_integers` to add a bunch efficiently, or `add_node_data`).

I don't think we have Exodus output support for it, though.  That was a hack Yaqi put in, I think he might only have put in the very specific case (element, integer, restricted range) he needed, and I've barely touched it since other than to make a couple use cases a little less precarious.
